src/query/graphite/native/aggregation_functions.go

```diff
@@ -561,24 +561,54 @@ func applyByNode(ctx *common.Context, seriesList singlePathSpec, nodeNum int, te
 //
 //    sumSeries(foo.by-function.server1.*.cpu.load5),sumSeries(foo.by-function.server2.*.cpu.load5),...
 func groupByNode(ctx *common.Context, series singlePathSpec, node int, fname string) (ts.SeriesList, error) {
-	metaSeries := make(map[string][]*ts.Series)
-	for _, s := range series.Values {
-		parts := strings.Split(s.Name(), ".")
+	return groupByNodes(ctx, series, fname, []int{node}...)
+}
+
+func findFirstMetricExpression(seriesName string) string {
+	idxOfRightParen := strings.Index(seriesName, ")")
+	substring := seriesName[:idxOfRightParen]
+	idxOfLeftParen := strings.LastIndex(substring, "(")
```

Comment:
Can you make `findFirstMetricExpression` potentially return an error if either `strings.Index` or `strings.LastIndex` return a `-1`?

Otherwise it will cause out of bounds panic, better to always optionally return an error rather than panic.

Reply:
Maybe return `(string, bool)` and if either of them return a -1 then the second bool param value returns `false`.

That way you can do:
```go
func getParts(series *ts.Series) {
  seriesName := series.Name()
  if metricExpr, ok := findFirstMetricExpression(seriesName); ok {
    seriesName = metricExpr
  }
  return strings.Split(seriesName, ".")
}
```

Reply:
Done! Great idea.